Gp/updatemacos

JackPiri opened the pull request:
This PR updates the macos image for Travis CI.
Some adaptations have been required mainly due to deprecated features.
Additionally, a minor refactoring has been provided for test `sc_cert_getblocktemplate.py`.

drgora reviewed (approved):
Looks good to me, thanks!

JackPiri committed:
Updated Travis MacOS image from v12.3 to v14.2

FIxes for std::allocator deprecation
Update to Boost 1.82
FIxes for sprintf deprecation
Silence warning deprecated-copy
Fixes for std::result_of deprecation
Refactoring of sc_cert_getblocktemplate.py
Removal of limitation of process memory usage (osx only)

a-petrini reviewed (approved):
Great PR, thanks!

titusen reviewed (approved):
LGTM, thanks.
